compiler/src/iree/compiler/Dialect/Flow/Transforms/AnnotateDispatches.cpp

```diff
@@ -160,12 +160,43 @@ static std::string getOpNameWithoutDialectName(Operation *op) {
 }
 
 static std::string summarizeLinalgOp(linalg::LinalgOp op) {
-  auto opName = op->getName().getStringRef();
-  if (!opName.consume_front("linalg."))
-    return "";
+  std::string prefix;
+
+  // Check if the op is a transpose and mark it as such for a better summary.
+  {
+    // Check if the body only contains a yield.
+    bool hasOnlyYield = true;
+    op.walk([&](Operation *child) {
+      if (op == child || isa<linalg::YieldOp>(child))
+        return WalkResult::advance();
+      hasOnlyYield = false;
+      return WalkResult::interrupt();
+    });
```

Comment:
The `yield` op must be the last op by design. We can maybe simplify it to `op.getBlock().without_terminator().empty()`.

Reply:
Fixed!